src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1243,6 +1233,85 @@ lExit: ;
 }
 FCIMPLEND
 
+static bool IsFastPathSupportedHelper(FieldDesc* pFieldDesc)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(pFieldDesc));
+    }
+    CONTRACTL_END;
+
+    return !pFieldDesc->IsThreadStatic() &&
+        !pFieldDesc->IsEnCNew() &&
+        !pFieldDesc->IsCollectible();
```

Comment:
```suggestion
        !(pFieldDesc->IsCollectible() && pFieldDesc->IsStatic())
```
Instance fields in collectible assemblies should be fine

Reply:
Sure. Thanks